app/controllers/concerns/saml_idp_auth_concern.rb

```diff
@@ -5,7 +5,7 @@ module SamlIdpAuthConcern
 
   included do
     # rubocop:disable Rails/LexicallyScopedActionFilter
-    before_action :validate_saml_request, only: :auth
+    before_action :create_saml_request_object, only: :auth
```

Comment:
renamed since the purpose of this method is really just to create the `saml_request` object. the `validate` naming was misleading.

Reply:
Is it technically "validating" in the sense that [the `head` called within `validate_saml_request`](https://github.com/18F/saml_idp/blob/272586f41eba7fe5074178bbcf206472f6ca82fc/lib/saml_idp/controller.rb#L21-L32) would terminate subsequent request handling if reached? The documentation is a little ambiguous in whether this as a ["renders or redirects"](https://api.rubyonrails.org/v7.1.3/classes/AbstractController/Callbacks/ClassMethods.html#method-i-before_action), but in some brief testing, it seems to?

Reply:
hmm yeah this is true! 

i think the current name feels misleading for a couple reasons:
- passing this validation doesn't mean that the request is valid, since other validations happen in the `SamlRequestValidator`. 
- as a side effect, it's making the `saml_request` object that is required for the other validations, which is very important and impossible to tell semantically

but yeah, my updated name is also misleading because it is doing some basic validation. i'll rename it!

Reply:
updated in https://github.com/18F/identity-idp/pull/9960/commits/777d3b5054634458f64062cba0189cbb009a58ea